put/request.go

```diff
@@ -582,3 +589,36 @@ func dirIsProjectOrTeamOrUsers(dir string) bool {
 func dirIsLustreWithPTUSubDir(dir string, ptuPart, numParts int) bool {
 	return dir == "lustre" && ptuPart >= 4 && ptuPart+2 <= numParts-1
 }
+
+// ValidateAndCreateUserMetadata takes a key=value string, validates it as a
+// metadata value then returns the key prefixed with the user namespace,
+// 'ibackup:user:', and the value. Returns an error if the meta is invalid.
+func ValidateAndCreateUserMetadata(kv string) (string, string, error) {
+	parts := strings.Split(kv, "=")
+	if len(parts) != validMetaParts {
+		return "", "", errInvalidMetaLength
+	}
+
+	key, err := handleNamespace(parts[0])
+	value := parts[1]
+
+	return key, value, err
+}
+
+// handleNamespace prefixes the user namespace 'ibackup:user:' onto the key if
+// it isn't already included. Returns an error if the key contains an invalid
+// namespace.
+func handleNamespace(key string) (string, error) {
+	keyParts := len(strings.Split(key, ":"))
+
+	switch {
+	case keyParts == 1:
+		return MetaUserNamespace + key, nil
+	case keyParts != validMetaWithNamespaceKeyParts:
+		return "", errInvalidMetaNamespace
+	case strings.Contains(key, MetaUserNamespace):
```

Comment:
Write a test that expects an error given "mykey:ibackup:user", then fix this so it does return an error.

Likewise, another test that should return an error is with key "ibackup:user:mykey:mysubkey". I think we shouldn't allow users to create their own sub-namespaces.